ariadne/cardano/src/Ariadne/Wallet/Backend/Restore.hs

```diff
@@ -117,20 +116,20 @@ restoreFromKeyFile acidDb face runCardanoMode mbWalletName path rType = do
 
 restoreFromSecretKey ::
        HasConfiguration
-    => AcidState DB
+    => PassiveWalletLayer IO
     -> WalletFace
     -> (CardanoMode ~> IO)
     -> Maybe WalletName
     -> EncryptedSecretKey
     -> WalletRestoreType
-    -> HasSpendingPassword
+    -> HasNonemptyPassphrase
     -> AssuranceLevel
     -> IO ()
-restoreFromSecretKey acidDb face runCardanoMode mbWalletName esk rType hasPass assurance = do
+restoreFromSecretKey pwl face runCardanoMode mbWalletName esk rType hasPP assurance = do
     utxoByAccount <- case rType of
         WalletRestoreQuick -> pure mempty
         WalletRestoreFull  -> runCardanoMode $ collectUtxo esk
-    addWallet acidDb face runCardanoMode esk mbWalletName utxoByAccount hasPass assurance
+    addWallet pwl face esk mbWalletName utxoByAccount hasPP assurance
```

Comment:
would it make some sense to turn all these functions into `ReaderT PassiveWalletLayer IO`? that'd increase the diff though...

Reply:
Definitely not in this PR.